docs/userguides/transactions.md

```diff
@@ -231,6 +231,34 @@ receipt = networks.provider.get_receipt(txn_hash)
 receipt.show_trace()
 ```
 
+## Gas Reports
+
+To view the gas report of a transaction receipt, use the  [`ReceiptAPI.show_gas_report()`](../methoddocs/api.html?highlight=receiptapi#ape.api.transactions.ReceiptAPI.show_gas_report) method:
+
+```python
+from ape import networks
+
+txn_hash = "0x053cba5c12172654d894f66d5670bab6215517a94189a9ffc09bc40a589ec04d"
+receipt = networks.provider.get_receipt(txn_hash)
+receipt.show_gas_report()
+```
+
+It will output tables of contracts and methods with gas usages that look like this:
+
+```bash
+                           DAI Gas
+┏━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━┳━━━━━━━┳━━━━━━━━┓
+┃ Method    ┃ Times called ┃ Min.  ┃ Max.  ┃ Mean  ┃ Median ┃
+┡━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━╇━━━━━━━╇━━━━━━━━┩
+│ balanceOf │ 4            │ 1302  │ 1302  │ 1302  │ 1302   │
+│ transfer  │ 5            │ 6974  │ 30374 │ 20174 │ 26174  │
+│ allowance │ 1            │ 1377  │ 1377  │ 1377  │ 1377   │
+│ approve   │ 1            │ 22414 │ 22414 │ 22414 │ 22414  │
+│ burn      │ 1            │ 11946 │ 11946 │ 11946 │ 11946  │
+│ mint      │ 1            │ 25845 │ 25845 │ 25845 │ 25845  │
```

Comment:
Does this only show methods called of the given target contract? or all call node targets inside of the call?

If the latter, it seems like a column for contract type would be necessary

Reply:
It is `1` table per contract, where the title of the contract is above the table. This table here is only for the `DAI `contract. Every target gets its own table. So if you had:

```
ContractA calls ContractB
ContractA calls ContractC
ContractC calls ContractB
```

, there would only be `1` table for `ContractB` containing data from both `ContractA` calling it as we al as `ContractC` calling it. And `ContractA` and `ContractC` would have their own tables with their own methods on it.

That is the intent anyway!

Reply:
I just did some manual testing:

If you run `ape run sim_gas` from this project: https://github.com/unparalleled-js/ape-trace-project, it calls a method in `ContractA` that calls a method in `ContractB` that calls a method in `ContractC. Then, it directly calls the method in `ContractC`. The resulting report looks like this:

```
                            ContractA Gas                             
┏━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━┳━━━━━━━┳━━━━━━━━┓
┃ Method             ┃ Times called ┃ Min.  ┃ Max.  ┃ Mean  ┃ Median ┃
┡━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━╇━━━━━━━╇━━━━━━━━┩
│ mergeGasReportTest │ 1            │ 13166 │ 13166 │ 13166 │ 13166  │
└────────────────────┴──────────────┴───────┴───────┴───────┴────────┘
                              SYMBOL Gas                              
┏━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━┳━━━━━━┳━━━━━━┳━━━━━━━━┓
┃ Method                ┃ Times called ┃ Min. ┃ Max. ┃ Mean ┃ Median ┃
┡━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━╇━━━━━━╇━━━━━━╇━━━━━━━━┩
│ callContractCMethodC3 │ 1            │ 5327 │ 5327 │ 5327 │ 5327   │
└───────────────────────┴──────────────┴──────┴──────┴──────┴────────┘
                      ContractC Gas                      
┏━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━┳━━━━━━┳━━━━━━┳━━━━━━━━┓
┃ Method   ┃ Times called ┃ Min. ┃ Max. ┃ Mean ┃ Median ┃
┡━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━╇━━━━━━╇━━━━━━╇━━━━━━━━┩
│ methodC3 │ 2            │ 166  │ 166  │ 166  │ 166    │
└──────────┴──────────────┴──────┴──────┴──────┴────────┘
```

so the resulting tables are per-contract as intended!